dubbo-kubernetes/src/main/java/org/apache/dubbo/registry/kubernetes/KubernetesServiceDiscovery.java

```diff
@@ -206,28 +211,43 @@ public void addServiceInstancesChangedListener(ServiceInstancesChangedListener l
     }
 
     private void watchEndpoints(ServiceInstancesChangedListener listener, String serviceName) {
-        Watch watch = kubernetesClient
-            .endpoints()
-            .inNamespace(namespace)
-            .withName(serviceName)
-            .watch(new Watcher<Endpoints>() {
-                @Override
-                public void eventReceived(Action action, Endpoints resource) {
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Received Endpoint Event. Event type: " + action.name() +
-                            ". Current pod name: " + currentHostname);
+        SharedIndexInformer<Endpoints> endInformer = kubernetesClient
+                .endpoints()
+                .inNamespace(namespace)
+                .withName(serviceName)
+                .inform(new ResourceEventHandler<Endpoints>() {
+                    @Override
+                    public void onAdd(Endpoints endpoints) {
+                        if (logger.isDebugEnabled()) {
+                            logger.debug("Received Endpoint Event. Event type: added. Current pod name: " + currentHostname +
+                                    ". Endpoints is: " + endpoints);
+                        }
+                        SERVICE_INSTANCE_MAP.put(getServiceCacheKey(serviceName), toServiceInstance(endpoints, serviceName));
```

Comment:
Must endpoints be cached, can `endpoints` be used directly by passing to `notifyServiceChanged`?

Reply:
I have removed useless SERVICE_INSTANCE_MAP. Pass serviceinstanceList which converted by endpoints directly to notifyServiceChanged.